mantle/kola/harness.go

```diff
@@ -436,6 +448,145 @@ func RunUpgradeTests(patterns []string, pltfrm, outputDir string, propagateTestE
 	return runProvidedTests(register.UpgradeTests, patterns, pltfrm, outputDir, propagateTestErrors)
 }
 
+func registerExternalTest(testname, executable, dependencydir, ignition string) error {
+	ignc3, _, err := ignv3.Parse([]byte(ignition))
+	if err != nil {
+		return errors.Wrapf(err, "Parsing config.ign")
+	}
+
+	unitname := "kola-runext.service"
+	remotepath := fmt.Sprintf("/usr/local/bin/kola-runext-%s", filepath.Base(executable))
+	unit := fmt.Sprintf(`[Unit]
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+Environment=%s=%s
+ExecStart=%s
+[Install]
+RequiredBy=multi-user.target
+`, kolaExtBinDataEnv, kolaExtBinDataDir, remotepath)
+	runextconfig := ignv3types.Config{
+		Ignition: ignv3types.Ignition{
+			Version: "3.0.0",
```

Comment:
Will this matter for RHCOS being on spec2x?

Reply:
Because this just hooks into the main `kola run` harness code, we reuse the [Ignition conversion](https://github.com/coreos/mantle/pull/1186) that already landed.